src/lib/bootloader/stage1.rb

```diff
@@ -144,33 +140,29 @@ def available_locations
       res
     end
 
-  private
-
-    def md_disk?(dev)
-      dev = Bootloader::UdevMapping.to_kernel_device(dev)
+    def can_use_boot?
+      tm = Yast::Storage.GetTargetMap
+      partition = Yast::BootStorage.BootPartitionDevice
 
-      disk_dev = Yast::Storage.GetDiskPartition(dev)
-      disk_dev = disk_dev["disk"] || ""
-      return false if disk_dev != dev # not a disk
+      part = Yast::Storage.GetPartition(tm, partition)
 
-      disk_map = Yast::Storage.GetTargetMap[dev]
-      disk_map && disk_map["type"] == :CT_MD
-    end
-
-    def md_disks
-      partitions = underlying_boot_partition_devices
-      disks = partitions.map do |part|
-        disk_dev = Yast::Storage.GetDiskPartition(part)
-        disk_dev["disk"]
+      if !part
+        log.error "cannot find partition #{partition}"
+        return false
       end
 
-      log.info "md_disks: #{disks.inspect} laying on #{partitions.inspect}"
+      log.info "Boot partition info #{part.inspect}"
 
-      disks
+      # cannot install stage one to xfs as it doesn't have reserved space (bnc#884255)
+      return false if part["used_fs"] == :xfs
+
+      true
```

Comment:
well, there can was more conditions in past, that is reason why I wrote it this way and I expect it will be extended in future.